blueflood-http/src/main/java/com/rackspacecloud/blueflood/http/HttpRequestWithDecodedQueryParams.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright 2013 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.http;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.DecoderResult;
+import io.netty.handler.codec.http.*;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class is a special type of {@link io.netty.handler.codec.http.HttpRequest}
+ * that knows how to decode the query parameters.
+ */
+public class HttpRequestWithDecodedQueryParams implements FullHttpRequest {
+
+    private Map<String, List<String>> queryParams;
+    protected FullHttpRequest request;
+
+    protected HttpRequestWithDecodedQueryParams(FullHttpRequest request, Map<String, List<String>> queryParams) {
```

Comment:
I am little confused here. If I create an instance of `HttpRequestWithDecodedQueryParams` and then using that instance, if I do `getQueryParams`, would I get decoded query params?